packages/superset-ui-chart/src/clients/ChartClient.ts

```diff
@@ -0,0 +1,137 @@
+import { isDefined } from '@superset-ui/core';
+import {
+  SupersetClient,
+  RequestConfig,
+  SupersetClientClass,
+  SupersetClientResponse,
+  Json,
+} from '@superset-ui/connection';
+import getChartBuildQueryRegistry from '../registries/ChartBuildQueryRegistrySingleton';
+import { FormData, AnnotationLayerMetadata } from '../query/FormData';
+
+interface SliceIdAndOrFormData {
+  sliceId?: number;
+  formData?: FormData;
+}
+
+interface AnnotationData {
+  [key: string]: object;
+}
+
+interface ChartData {
+  annotationData: AnnotationData;
+  datasource: object;
+  formData: FormData;
+  queryData: object;
+}
+
+export interface ChartClientConfig {
+  client?: SupersetClientClass;
+}
+
+export class ChartClient {
+  readonly client: {
+    get: (config: RequestConfig) => Promise<SupersetClientResponse>;
+    post: (config: RequestConfig) => Promise<SupersetClientResponse>;
+  };
+
+  constructor(config: ChartClientConfig = {}) {
+    const { client = SupersetClient } = config;
+    this.client = client;
+  }
+
+  loadFormData(input: SliceIdAndOrFormData, options?: RequestConfig): Promise<FormData> {
+    /* If sliceId is provided, use it to fetch stored formData from API */
+    if (input.sliceId) {
+      const promise = this.client
+        .get({
+          endpoint: `/api/v1/formData/?slice_id=${input.sliceId}`,
+          ...options,
+        } as RequestConfig)
+        .then(response => response.json as Json);
+
+      /*
+       * If formData is also specified, override API result
+       * with user-specified formData
+       */
+      return input.formData
+        ? promise.then(
+            (dbFormData: object) =>
```

Comment:
Think we need to destruct the `json` field here to get the form data json.

Reply:
never mind. I check the response of `slice_json` and found what you mean